pkg/webui/components/key-value-map/entry.js

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import bind from 'autobind-decorator'
+import { defineMessages } from 'react-intl'
+
+import PropTypes from '../../lib/prop-types'
+import Input from '../input'
+import Button from '../button'
+
+import style from './key-value-map.styl'
+
+const m = defineMessages({
+  deleteEntry: 'Delete Entry',
+})
+
+@bind
+class Entry extends React.Component {
+
+  handleRemoveButtonClicked (event) {
+    const { onRemoveButtonClick, index } = this.props
+    onRemoveButtonClick(index, event)
+  }
+
+  handleKeyChanged (newKey) {
+    const { onChange, index } = this.props
+    onChange(index, { key: newKey })
+  }
+
+  handleValueChanged (newValue) {
+    const { onChange, index } = this.props
+    onChange(index, { value: newValue })
+  }
+
+  render () {
+    const {
+      name,
+      index,
+      keyPlaceholder,
+      valuePlaceholder,
+      value,
+      onBlur,
+    } = this.props
+
+    return (
+      <div className={style.entriesRow} key={index}>
+        <Input
+          className={style.input}
+          name={`${name}[${index}].key`}
+          placeholder={keyPlaceholder}
+          type="text"
+          onChange={this.handleKeyChanged}
+          value={value.key}
+        />
+        <Input
+          className={style.input}
+          name={`${name}[${index}].value`}
+          placeholder={valuePlaceholder}
+          type="text"
+          onChange={this.handleValueChanged}
+          onBlur={onBlur}
+          value={value.value}
+        />
+        <Button
+          name={`${name}[${index}].remove`}
```

Comment:
do you use the name anywhere? you already pass a separate event handler for the button, why adding the name?